### PR TITLE
Policies: Request circulation policies

### PR DESCRIPTION
**Pull request by BadrAly**
* NEW Adds request circulation policies

Co-authored-by: Aly Badr <aly.badr@rero.ch>
Co-authored-by: bertrand Zuchuat <bertrand.zuchuat@rero.ch>

### CODE REVIEW

File: invenio_circulation/config.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
     PendingToItemInTransitPickup, ToItemOnLoan
 from .utils import get_default_extension_duration, \
     get_default_extension_max_count, get_default_loan_duration, \
-    is_item_available, is_loan_duration_valid, item_exists, \
+    is_can_request, is_item_available, is_loan_duration_valid, item_exists, \
```
Comment (zzacharo) [inline invenio_circulation/config.py]: Probably `can_be_requested`?

File: invenio_circulation/utils.py [hunk 2]
```diff
@@ -60,6 +60,16 @@ def is_item_available(item_pid):
     )
 
 
+def is_can_request(record_pid=None, request_level=None):
```
Comment (zzacharo) [inline invenio_circulation/utils.py]: Same here.

File: invenio_circulation/errors.py [hunk 3]
```diff
@@ -25,6 +25,10 @@ class ItemNotAvailable(CirculationException):
     """Exception raised from action on unavailable item."""
 
 
+class RecordCanNotSatisfyAction(CirculationException):
```
Comment (zzacharo) [inline invenio_circulation/errors.py]: Probably `RecordCannotBeRequested`?

**Review by zzacharo — CHANGES_REQUESTED** [review]

**Comment by ntarocco** [comment]
@BadrAly I want to give you our point of view on this, because we had similar thoughts.

The `CREATED` status is a "fake" one: we need a starting point to perform the first transition to `REQUESTED` or to another state, but ideally we should never have loans in state `CREATED`.

Are you planning to build a UI to find all CREATED and clean them up at certain point?

In our case, we decided that we never want this state, so we actually check in ILS, before, if the Item can be requested and then we perform in one single atomic operation:
```
# check first if loan can be created and requested for the given item
loan = Loan.create()  # status will be CREATED
circulation.trigger(
    loan, **dict(params, trigger="request")
)
# status will be REQUESTED
```
Basically for us is or `REQUESTED`, or it does not exist in our db.

**Comment by BadrAly** [comment]
@ntarocco : Could you please merge this PR?
We already manage the  `CREATED` status as you explained on our UI. @jma will add more to this later.

**Review by ntarocco — APPROVED** [review]
